spec/gameSpec.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+describe('Game', function() {
+
+  var game;
+
+  beforeEach(function() {
+    game = new Game();
+  });
+
+  it('starts with a total score of zero', function() {
+    expect(game.totalScore).toEqual(0);
+  });
+
+  it('starts a new frame after two rolls', function() {
+    game.recordRoll(5);
+    game.recordRoll(4);
+    expect(game.currentFrame.rolls).toEqual([]);
+    expect(game.allFrames[0].rolls).toEqual([5,4]);
+  });
+
+  it('starts a new frame after a strike', function() {
+    game.recordRoll(10);
+    expect(game.currentFrame.rolls).toEqual([]);
+    expect(game.allFrames[0].rolls).toEqual([10]);
+  });
+
+  it('increases the frame number by one',function() {
+    game.recordRoll(10)
+    expect(game.frameNum).toEqual(2);
+  });
+
+  describe('Bonus Scores', function() {
+
+    beforeEach(function() {
+      game.recordRoll(9);
+      game.recordRoll(1);
+      game.recordRoll(3);
+      game.addSpareBonus();
```

Comment:
It should not be necessary for me to call `addSpareBonus()` manually - the game and frames should handle that automatically.